client/my-sites/site-settings/form-general.jsx

```diff
@@ -286,48 +370,50 @@ const FormGeneral = React.createClass( {
 						<FormRadio
 							name="blog_public"
 							value="-1"
-							checked={ -1 === parseInt( this.state.blog_public, 10 ) }
+							checked={ -1 === parseInt( fields.blog_public, 10 ) }
 							onChange={ this.handleRadio }
-							disabled={ this.state.fetchingSettings }
+							disabled={ isRequestingSettings }
 							onClick={ this.onRecordEvent( 'Clicked Site Visibility Radio Button' ) } />
-						<span>{ this.translate( 'Private' ) }</span>
+						<span>{ translate( 'Private' ) }</span>
 						<FormSettingExplanation isIndented>
-							{ this.translate( 'Your site is only visible to you and users you approve.' ) }
+							{ translate( 'Your site is only visible to you and users you approve.' ) }
 						</FormSettingExplanation>
 					</FormLabel>
 				}
 
 			</FormFieldset>
 		);
-	},
+	}
 
-	handleAmpToggle() {
-		this.setState( { amp_is_enabled: ! this.state.amp_is_enabled }, () => {
+	handleAmpToggle = () => {
+		const { fields, updateFields } = this.props;
+		updateFields( { amp_is_enabled: ! fields.amp_is_enabled }, () => {
```

Comment:
`updateFields` only accepts one argument and cannot be called in its current form with the `setState` callback. You'd need to add this to the higher-order component if that's the behavior you're looking for.